openlibrary/plugins/openlibrary/js/autocomplete.js

```diff
@@ -16,59 +57,82 @@ export default function($) {
-                if (ol_ac_opts.addnew == true || (ol_ac_opts.addnew && ol_ac_opts.addnew(query))) {
-                    parsed = parsed.slice(0, ac_opts.max - 1);
-                    const name = ol_ac_opts.new_name || query;
-                    parsed.push({
-                        data: {name, key: '__new__'},
-                        value: name,
-                        result: name
-                    });
-                }
-                return parsed;
             },
-        };
-
-        $(_this)
-            .autocomplete(ol_ac_opts.endpoint, $.extend(default_ac_opts, ac_opts))
-            .result(function(event, item) {
-                var $this;
-
-                $(`#${this.id}-key`).val(item.key);
-                $this = $(this);
-
-                //adding class directly is not working when tab is pressed. setTimeout seems to be working!
+            select: function (_event, ui) {
+                var item = ui.item;
+                var $this = $(this);
+                $(`#${_this.id}-key`).val(item.key);
                 setTimeout(function() {
                     $this.addClass('accept');
                 }, 0);
-            })
-            .nomatch(function(){
-                $(`#${this.id}-key`).val('');
-                $(this).addClass('reject');
-            })
-            .keypress(function() {
+            },
+            mustMatch: true,
+            formatMatch: function(item) { return item.name; }
+        };
+
+        $.widget('custom.autocompleteHTML', $.ui.autocomplete, {
+            _renderMenu($ul, items) {
+                $ul.addClass('ac_results').attr('id', this.ulRef);
+                items.forEach((item) => {
+                    $('<li>')
+                        .attr('data-value', item.value)
+                        .data('ui-autocomplete-item', item)
+                        .attr('aria-label', item.value)
+                        .html(item.label)
+                        .appendTo($ul);
+                });
+                // store list so we can add ac_over hover effect in `focus` event
+                $(_this).data('list', $ul);
+            }
+        });
+
+        const options = $.extend(default_ac_opts, ac_opts, {
+            source: function (q, response) {
+                const term = q.term;
+                return $.ajax({
+                    url: ol_ac_opts.endpoint,
+                    data: {
+                        q: term,
+                        limit: options.max,
+                        timestamp: new Date()
+                    }
+                }).then((results) => {
+                    response(
+                        mapApiResultsToAutocompleteSuggestions(
+                            results,
+                            (r) => highlight(options.formatItem(r), term),
+                            ol_ac_opts.addnew === true ||
+                                (ol_ac_opts.addnew && ol_ac_opts.addnew(term)) ? term : null
```

Comment:
Fixes the asterisk point

```suggestion
                            ol_ac_opts.addnew === true ||
                                (ol_ac_opts.addnew && ol_ac_opts.addnew(term)) ? (ol_ac_opts.new_name || term) : null
```